vault/plugin_catalog.go

```diff
@@ -59,12 +84,98 @@ func (c *Core) setupPluginCatalog(ctx context.Context) error {
 	return nil
 }
 
+func (c *PluginCatalog) removeMultiplexedClient(ctx context.Context, name, id string) {
+	if _, ok := c.multiplexedClients[name]; !ok {
```

Comment:
Do we need any locking on `c.multiplexedClients`? (I've just started looking at the code and maybe the locks, if needed, are elsewhere.)

Reply:
Yes, it most likely will need locks. I haven't added any around this object yet.

Reply:
Thinking about this a bit more, I think the lock for PluginCatalog should be sufficient. So I don't think any locking is needed on c.multiplexedClients.

Reply:
addressed in https://github.com/hashicorp/vault/pull/13734/commits/b9d185e11083094629bd6a60bdc37d7ed573edc2